subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DefaultTransformerInvocationFactory.java

```diff
@@ -376,7 +376,7 @@ public void visitIdentityInputs(InputVisitor visitor) {
         @Override
         @OverridingMethodsMustInvokeSuper
         public void visitRegularInputs(InputVisitor visitor) {
-            visitor.visitInputFileProperty(INPUT_ARTIFACT_PROPERTY_NAME, PRIMARY,
+            visitor.visitInputFileProperty(INPUT_ARTIFACT_PROPERTY_NAME, INCREMENTAL,
```

Comment:
IIUC, this changes the behavior from the primary input always being `@SkipWhenEmpty` to never being `@SkipWhenEmpty`. Would it make sense for some transforms to allow marking their primary input as `@SkipWhenEmpty` even if the default is the opposite?

Reply:
It doesn't change the behavior, since the execution engine didn't handle skip when empty before. So transforms never supported this. It slipped in accidentally with the other PR.

Reply:
I see. Still, do we want to (eventually) support `@SkipWhenEmpty` for primary transform inputs? If so, I think we should add an issue for this. Otherwise let's ship this.

Reply:
> Still, do we want to (eventually) support @SkipWhenEmpty for primary transform inputs?

We could, that would mean that we would create an empty results file (aka empty list of files) when the sources are empty. Though I can't come up with a good use-case now, so I wouldn't even add an issue for it.